### PR TITLE
[wip] Epic: Cluster-Wide to GA

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

File: pkg/controller/servicemesh/member/namespace_reconciler.go [hunk 1]
```diff
@@ -236,10 +238,13 @@ func (r *namespaceReconciler) reconcileNamespaceInMesh(ctx context.Context, name
 
 	allErrors := []error{}
 
-	// add role bindings
-	err = r.reconcileRoleBindings(ctx, namespace)
-	if err != nil {
-		allErrors = append(allErrors, err)
+	// if cluster-wide, v2.4+ SMCPs, istiod uses ClusterRoleBindings, so there's no need to create RoleBindings in the member namespace
+	if r.meshVersion.LessThan(versions.V2_4) || !r.clusterWideMode {
```
Comment (dgn) [inline pkg/controller/servicemesh/member/namespace_reconciler.go:242]: tiny nit: the bool comparison `!r.clusterWideMode` is going to be faster, so maybe we should do that first?

File: pkg/controller/servicemesh/memberroll/controller.go [hunk 2]
```diff
@@ -354,21 +345,27 @@ func (r *MemberRollReconciler) reconcileObject(ctx context.Context, roll *maistr
...
+		if roll.MatchesNamespacesDynamically() {
+			panic("namespace wildcards & selectors for Kiali are not yet implemented")
 		}
 
-		if meshIsClusterScoped && roll.Spec.IsClusterScoped() {
-			kialiErr = r.kialiReconciler.reconcileKiali(ctx, kialiName, meshNamespace, []string{"**"}, getExcludedNamespaces())
-		} else {
-			kialiErr = r.kialiReconciler.reconcileKiali(ctx, kialiName, meshNamespace, allKnownMembers.List(), []string{})
-		}
+		// var version versions.Version
```
Comment (dgn) [inline pkg/controller/servicemesh/memberroll/controller.go:352]: commented out code

File: pkg/controller/versions/strategy_v2_4.go [hunk 3]
```diff
@@ -681,3 +689,41 @@ func (v *versionStrategyV2_4) validateGlobal(ctx context.Context, version Ver, m
...
+						Kind:       "ServiceMeshControlPlane",
+						Name:       smcp.Name,
+						UID:        smcp.UID,
+						Controller: pointer.BoolPtr(true),
+					},
+				},
+			},
+			Spec: v1.ServiceMeshMemberRollSpec{
+				Members: []string{"*"},
+			},
```
Comment (dgn) [inline pkg/controller/versions/strategy_v2_4.go:719]: shouldn't we also exclude system namespaces here?

File: pkg/controller/servicemesh/memberroll/controller.go [hunk 4]
```diff
@@ -354,21 +345,27 @@ func (r *MemberRollReconciler) reconcileObject(ctx context.Context, roll *maistr
...
-		var version versions.Version
-		version, err = versions.ParseVersion(mesh.Spec.Version)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-		meshIsClusterScoped, err := version.Strategy().IsClusterScoped(&mesh.Spec)
-		if err != nil {
-			return reconcile.Result{}, err
+		if roll.MatchesNamespacesDynamically() {
+			panic("namespace wildcards & selectors for Kiali are not yet implemented")
```
Comment (dgn) [inline pkg/controller/servicemesh/memberroll/controller.go:349]: panicking doesn't look like a good choice? why not emit a warning and select all namespaces for now?

**Review by dgn — COMMENTED** [review]
sorry I only realized in the middle of the review that this is still WIP. will leave my comments here and come back later

**Comment by luksa** [comment]
Closing. Replaced by individual PRs.

**Pull request closed without merging**